Ricochet/Assets/_Scripts/UI/_UICharacterSelectModel.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace PlayerSelectData
+{
+
+    public class PlayerData
+    {
+        public bool active = false;
+        public int team = 1;
+        public bool selected = false;
+        public string character = "bean";
+        public int xPos = 0;
+        public int yPos = 0;
+        public bool ready = false;
+    }
+
+    public class _UICharacterSelectModel : MonoBehaviour
+    {
+        #region Inspector Variables
+        [Header("Default settings")]
+        [Tooltip("Default number of teams")]
+        [SerializeField]
+        private int numberOfTeams = 2;
+        [Tooltip("Whether players need to be on both sides to play a game(for testing purposes)")]
+        [SerializeField]
+        private bool validationNeeded = true;
+
+        [Header("Player UI Objects")]
+        [Tooltip("Player 1")]
+        [SerializeField]
+        private _UICharacterSelectView player1;
+        [Tooltip("Player 2")]
+        [SerializeField]
+        private _UICharacterSelectView player2;
+        [Tooltip("Player 3")]
+        [SerializeField]
+        private _UICharacterSelectView player3;
+        [Tooltip("Player 4")]
+        [SerializeField]
+        private _UICharacterSelectView player4;
+        #endregion
+
+        #region Private Variables
+        private Dictionary<int, PlayerData> players;
+        private Dictionary<int, _UICharacterSelectView> spriteHandlers;
+        private string[,] characters = { { "bean", "carrot" } };
+        private int numberOfPlayers = 4;
+        private GameManager gameManagerInstance;
+        #endregion
+
+        #region MonoBehaviour
+        void Awake()
+        {
+            players = new Dictionary<int, PlayerData>();
+            for (int i = 1; i <= numberOfPlayers; i++)
+            {
+                players.Add(i, new PlayerData());
+            }
+            spriteHandlers = new Dictionary<int, _UICharacterSelectView>()
+        {
+            {1, player1},
+            {2, player2},
+            {3, player3},
+            {4, player4}
+        };
+        }
+        #endregion
+
+        public bool ValidateSetup()
+        {
+            bool team1 = false;
+            bool team2 = false;
+            foreach (PlayerData player in players.Values)
+            {
+                if (player.active)
+                {
+                    if (!player.ready)
+                    {
+                        return false;
+                    }
+                    else if (player.team == 1)
+                    {
+                        team1 = true;
+                    }
+                    else if (player.team == 2)
+                    {
+                        team2 = true;
+                    }
+                }
+            }
+            if (validationNeeded)
```

Comment:
This "validationNeeded" check could be done before iterating through all the players

Reply:
Actually, nevermind. Just saw what the else statement did